cluster-autoscaler/core/scale_down.go

```diff
@@ -763,12 +763,12 @@ func drainNode(node *apiv1.Node, pods []*apiv1.Pod, client kube_client.Interface
 		allGone = true
 		for _, pod := range pods {
 			podreturned, err := client.CoreV1().Pods(pod.Namespace).Get(pod.Name, metav1.GetOptions{})
-			if err == nil {
+			if err == nil && podreturned.Spec.NodeName == node.Name {
```

Comment:
I will feel better if you add `podreturned != nil` check. It should probably never happen without some err, but it's still a pointer.